beamer/tests/agent/multiple_processors/test_multiple_event_processors.py

```diff
@@ -0,0 +1,241 @@
+import contextlib
+import json
+import os
+import signal
+import time
+from pathlib import Path
+from subprocess import DEVNULL, PIPE
+from typing import Any
+
+import ape
+import eth_account
+import psutil
+import yaml
+from eth_utils import to_canonical_address
+from yaml.loader import SafeLoader
+
+import beamer.agent.metrics
+from beamer.agent.agent import Agent
+from beamer.agent.config import Config
+from beamer.agent.contracts import ContractInfo
+from beamer.agent.typing import BlockNumber, ChainId
+from beamer.agent.util import TokenChecker, make_web3, transact
+from beamer.tests.util import Sleeper
+
+SLAVE_TEST_PATH = Path(__file__).parent / "test_slave.py"
+CONFIG_PATH = os.path.join(os.getcwd(), "ape-config.yaml")
+BACKUP_PATH = os.path.join(os.getcwd(), "ape-config.yaml.2")
+
+
+# ape local account, to be used for fulfilling requests.
+# The private key here corresponds to the 10th account ganache creates on
+# startup.
+_LOCAL_ACCOUNT = ape.accounts.test_accounts[-1]
+
+
+def _create_and_return_backup_config() -> dict[str, Any]:
+    with open(CONFIG_PATH) as _f:
+        current_config = yaml.load(_f, Loader=SafeLoader)
+    with open(BACKUP_PATH, "w") as _f:
+        yaml.dump(current_config, _f, sort_keys=False, default_flow_style=False)
+    return current_config
+
+
+def _create_new_config(port: int):
+    current_config = _create_and_return_backup_config()
+    current_config["ganache"]["server"]["port"] = port
+    with open(CONFIG_PATH, "w") as _f:
+        yaml.dump(current_config, _f, sort_keys=False, default_flow_style=False)
+
+
+def _restore_backup_config(config: dict[str, Any]):
+    os.remove(CONFIG_PATH)
+    os.remove(BACKUP_PATH)
+    with open(CONFIG_PATH, "w") as _f:
+        yaml.dump(config, _f, sort_keys=False, default_flow_style=False)
```

Comment:
I suggest we replace

- `_create_and_return_backup_config` 
- `_create_new_config`
- `_restore_backup_config`

with just a single single function, a context manager that overwrites ape config, e.g.:

```python
   with _configure_ganache_port(port):
         slave = _start_slave_test()
         ...
```
where the context manager would take care to restore the original config.